pkg/apis/cluster/v1alpha1/types.go

```diff
@@ -116,6 +137,79 @@ type ClusterSpec struct {
 	// any resource that does not tolerate the Taint.
 	// +optional
 	Taints []corev1.Taint `json:"taints,omitempty"`
+
+	// ResourceModels is the list of resource modeling in this cluster. Each modeling quota can be customized by the user.
+	// Modeling name must be one of the following: cpu, memory, storage, ephemeral-storage.
+	// If the user does not define the modeling name and modeling quota, it will be the default model.
+	// The default model index from 0 to 11. Each default model's cpu quota is (2^index-1, 2^index], index >= 0
+	// Each default model's memory quota is also (2^index-1, 2^index], index >= 0, for example, model0 is like this:
+	// - index: 0
+	// ranges:
+	//   - name: "cpu"
+	//     min: 0 C
+	//     max: 1 C
+	//   - name: "memory"
+	//     min: 0 GB
+	//     max: 1 GB
+	//
+	// model10 is like this:
+	// - index: 10
+	// range:
+	//   - name: "cpu"
+	//     min: 512 C
+	//     max: 1024 C
+	//   - name: "memory"
+	//     min: 512 GB
+	//     max: 1024 GB
+	//
+	// model11 is like this:
+	// - index: 11
+	// range:
+	//   - name: "cpu"
+	//     min: 1024 C
+	//     max: MAXINT
+	//   - name: "memory"
+	//     min: 1024 GB
+	//     max: MAXINT
+	// +optional
+	ResourceModels []ResourceModel `json:"resourceModels,omitempty"`
+}
+
+// ResourceModel describes the modeling that you want to statistics.
+type ResourceModel struct {
+	// Grade is the index for the resource modeling.
+	// +optional
+	Grade int `json:"grade,omitempty"`
+
+	// Ranges describes the resource quota ranges.
+	// +optional
+	Ranges []ResourceModelItem `json:"ranges,omitempty"`
+}
+
+// ResourceModelItem describes the detail of each modeling quota that ranges from min to max.
+// Please pay attention, by default, the value of min can be obtained, and the value of max cannot be obtained.
+// For example, in an interval, min = 2, max =10 is set, which means the interval [2,10).
+// This rule ensure that all intervals have the same meaning. If the last interval is +∞,
+// it is definitely unreachable. Therefore, we define the right interval as the open interval.
+// For a valid interval, the value on the right is greater than the value on the left,
+// in other words, max must be greater than min.
```

Comment:
I'm confused by the description.
> by default, the value of min can be obtained, and the value of max cannot be obtained.

How we can `obtain` the min value? Why we can't obtain the `max` value?

By the way, you are using `[2, 10)`(left open, right close) which violates the description on `ResourceModels`.

Reply:
The `obtain` here may be inaccurate. Using `include` is more appropriate.

`[2, 10)` is (left close, right open).

The reason why makes a rule that the left interval is inclusive and the right interval is not inclusive. It's because of wanting to unify all situations. If the last interval contains +∞. The right interval is definitely not included. Therefore, it makes a rule that all intervals on the right interval are not included. This makes the program logic more unified.